redfish-core/lib/pcie.hpp

```diff
@@ -712,40 +712,34 @@ inline void requestRoutesSystemPCIeDevice(App& app)
             [](const crow::Request& req,
                const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
                const std::string& device) {
-        std::optional<std::vector<nlohmann::json>> slotsData;
-        if (!json_util::readJsonPatch(req, asyncResp->res, "Oem", slotsData))
+        std::optional<nlohmann::json> oem;
+        if (!json_util::readJsonPatch(req, asyncResp->res, "Oem", oem))
         {
             return;
         }
-        if (!slotsData || slotsData->empty())
+        if (!oem || oem->empty())
```

Comment:
The  oem->empty() made sense when slotsData was a vector but I don't think here it is needed

Reply:
Those invalid requests give 400.

For example,
```
redfishtool raw PATCH -r ${bmc}  -u service -p 0penBmc0 -S Always redfishtool raw PATCH -r ${bmc}  -u service -p 0penBmc0 -S Always /redfish/v1/Systems/system/PCIeDevices/chassis_motherboard_pcieslot8_pcie_card8  -d '{"Oem": {"IBM": {}}}'
   redfishtool: Transport: Response Error: status_code: 400 -- Bad Request
...
redfishtool raw PATCH -r ${bmc}  -u service -p 0penBmc0 -S Always /redfish/v1/Systems/system/PCIeDevices/chassis_motherboard_pcieslot8_pcie_card8  -d '{"Oem": {}}'
   redfishtool: Transport: Response Error: status_code: 400 -- Bad Request
```
I'll check out whether the checks like `if(!ibmOem)` is needed.

Reply:
So I fixed the code by adding those check and return error msg like

```
+        if (!ibmOem || ibmOem->empty())
+        {
+            messages::propertyMissing(asyncResp->res, "IBM");
+            return;
+        }
+
```